app/validators/email_validator.rb

```diff
@@ -0,0 +1,9 @@
+# app/validators/email_validator.rb
+
+class EmailValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    unless value =~ /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i

```

Comment:
Great, thanks. Changed to use Michael Hartl solution. Not quite good using regex, is something i have to work more.